compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUUnsupportedScalabilityToLoops.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TileSizeSelection.h"
+#include "iree/compiler/Codegen/LLVMCPU/PassDetail.h"
+#include "iree/compiler/Codegen/LLVMCPU/Passes.h"
+#include "iree/compiler/Codegen/LLVMCPU/Utils.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "mlir/Dialect/SCF/Utils/Utils.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-llvmcpu-unsupported-scalability-to-loops"
+#define VEC_DBGS() (llvm::dbgs() << '[' << DEBUG_TYPE << "] ")
+
+namespace mlir::iree_compiler {
+
+namespace {
+
+class LLVMCPUUnsupportedScalabilityToLoopsPass
+    : public LLVMCPUUnsupportedScalabilityToLoopsBase<
+          LLVMCPUUnsupportedScalabilityToLoopsPass> {
+public:
+  using LLVMCPUUnsupportedScalabilityToLoopsBase::
+      LLVMCPUUnsupportedScalabilityToLoopsBase;
+
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry
+        .insert<arith::ArithDialect, linalg::LinalgDialect, scf::SCFDialect>();
+  }
+
+  void runOnOperation() override;
+};
+
+static bool opKnownToSupport2DScalableVectorizationWithArmSME(Operation *op) {
+  return isa<linalg::MatmulOp, linalg::MatmulTransposeAOp, linalg::FillOp>(op);
+}
+
+struct DropUnsupportedScalableDimsFromTilingInterfaceOps
+    : public OpInterfaceRewritePattern<TilingInterface> {
+  DropUnsupportedScalableDimsFromTilingInterfaceOps(MLIRContext *context,
+                                                    bool assumeArmSME)
+      : OpInterfaceRewritePattern(context), assumeArmSME(assumeArmSME) {}
+
+  LogicalResult matchAndRewrite(TilingInterface op,
+                                PatternRewriter &rewriter) const override {
+    // Note: This rewrite is currently only required for ArmSME (which is the
+    // only target that currently has some concept of 2D scalability).
```

Comment:
IMHO this comment fits better in [compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.td](https://github.com/openxla/iree/pull/16712/files#diff-1e3b594a934d4a19e8f736d21de5e0a3f2e121b629a06a26247265c6d30089e4) - it's a general comment on the applicability of this pass. Definitely worth adding a note like this at the top of this file.